### PR TITLE
Support deploying Windows container images.

### PR DESCRIPTION
**Pull request by chunye**

### CODE REVIEW

**Review by jvano — APPROVED** [review]
:shipit:

**Comment by jvano** [comment]
@AmrutaKawade can you help us merging this PR?

**Review by pagariyaalok — APPROVED** [review]
:shipit:

File: packages/appservice-rest/src/Utilities/KuduServiceUtility.ts [hunk 1]
```diff
@@ -201,11 +201,10 @@ export class KuduServiceUtility {
...
         try {
             core.debug(`DeployWebAppImage - appName: ${appName}; images: ${images}; isLinux:${isLinux}`);
-            if (!isLinux) {
-                throw new Error("Windows Containerized web app is not available for Publish profile auth scheme.");
-            }
             console.log(`Deploying image ${images} to App Service ${appName}`);
-            let headers = {'LinuxFxVersion': `DOCKER|${images}`};
+
+            let fxVersionName = isLinux ? 'LinuxFxVersion' : 'WindowsFxVersion';
+            let headers = {fxVersionName: `DOCKER|${images}`};
```
Comment (pagariyaalok) [inline packages/appservice-rest/src/Utilities/KuduServiceUtility.ts:207]: >images [](http://example.com/codeflow?start=52&length=6)

Add check to confirm this is not empty ?

**Review by AmrutaKawade — COMMENTED** [review]
remove this validation https://github.com/Azure/webapps-deploy/blob/89bdb9149540d2ce35d43cfe13f5a6af4041c46b/src/ActionInputValidator/Validations.ts#L114
since this method is used in publishprofile container validator. It will not get called for windows based apps because of this validation

**Review by ds-ms — APPROVED** [review]

**Comment by JasonFreeberg** [comment]
@ds-ms @AmrutaKawade -- Quick question, will the `webapps-deploy` action need an update to use the latest version of this library?

File: packages/appservice-rest/src/Utilities/KuduServiceUtility.ts (hunk 1 above)
Comment (chunye) [reply packages/appservice-rest/src/Utilities/KuduServiceUtility.ts:211]: Done.

**Comment by AmrutaKawade** [comment]
> @ds-ms @AmrutaKawade -- Quick question, will the `webapps-deploy` action need an update to use the latest version of this library?

yes

File: packages/appservice-rest/src/Utilities/KuduServiceUtility.ts [hunk 2]
```diff
@@ -201,11 +201,14 @@ export class KuduServiceUtility {
     public async deployWebAppImage(appName: string, images: string, isLinux: boolean) {
         try {
             core.debug(`DeployWebAppImage - appName: ${appName}; images: ${images}; isLinux:${isLinux}`);
-            if (!isLinux) {
-                throw new Error("Windows Containerized web app is not available for Publish profile auth scheme.");
-            }
             console.log(`Deploying image ${images} to App Service ${appName}`);
-            let headers = {'LinuxFxVersion': `DOCKER|${images}`};
+
```
Comment (AmrutaKawade) [inline packages/appservice-rest/src/Utilities/KuduServiceUtility.ts:205]: Please upgrade version in package.json https://github.com/microsoft/pipelines-appservice-lib/blob/1af035a49f04cdc63ed8fbd5790b235a32ec03a7/packages/appservice-rest/package.json

Comment (chunye) [reply packages/appservice-rest/src/Utilities/KuduServiceUtility.ts:205]: Updated to 1.3.2.

File: packages/appservice-rest/src/Utilities/KuduServiceUtility.ts (hunk 1 above)
Comment (pagariyaalok) [reply packages/appservice-rest/src/Utilities/KuduServiceUtility.ts:211]: This might not check if images is string.empty.

**Comment by chunye** [comment]
@BALAGA-GAYATRI, can you approve?

**Review by AmrutaKawade — APPROVED** [review]

**Review by BALAGA-GAYATRI — APPROVED** [review]

**Review by BALAGA-GAYATRI — APPROVED** [review]

**Pull request merged**